src/auslib/services/releases.py

```diff
@@ -0,0 +1,379 @@
+import logging
+from copy import deepcopy
+from itertools import chain
+
+from deepmerge import Merger
+from flask import current_app as app
+from sqlalchemy import join, select
+
+from ..blobs.base import createBlob
+from ..errors import PermissionDeniedError, ReadOnlyError, SignoffRequiredError
+from ..global_state import dbo
+from ..util.data_structures import ensure_path_exists, get_by_path, infinite_defaultdict, set_by_path
+
+log = logging.getLogger(__file__)
+
+store = object()
+
+# fmt: off
+APP_RELEASE_ASSETS = {
+    "platforms": {
+        "*": {
+            "locales": {
+                "*": store
+            }
+        }
+    }
+}
+RELEASE_BLOB_ASSETS = {
+    1: APP_RELEASE_ASSETS,
+    2: APP_RELEASE_ASSETS,
+    3: APP_RELEASE_ASSETS,
+    4: APP_RELEASE_ASSETS,
+    5: APP_RELEASE_ASSETS,
+    6: APP_RELEASE_ASSETS,
+    7: APP_RELEASE_ASSETS,
+    8: APP_RELEASE_ASSETS,
+    9: APP_RELEASE_ASSETS,
+}
+# fmt: on
+
+
+release_merger = Merger(
+    [
+        # Merge dictionaries
+        (dict, ["merge"])
```

Comment:
That's a good point. Nightlies are intended to be in scope here, so we obviously need to get that right. I'll add some tests for this, and otherwise make sure it won't be an issue.

Reply:
After some consideration and discussion, I think the right thing to do is to have POSTs overwrite `completes` and `partials` (if they are passed). We don't have a known use case for wanting to append to them, and the majority of the people I asked thought overwriting was the more obvious behaviour anyways.

This is actually what happens already, but I'll be adding a test for it.